File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/angularapp/src/app/services/feedback.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Feedback } from '../models/feedback.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {
  apiUrl = "https://8080-fadbadfaddcabbbeddcbfaafaaacccecffb.premiumproject.examly.io/api/feedback";
  constructor(private http: HttpClient) {
  }
  sendFeedback(feedback: Feedback): Observable<Feedback> {
    console.log(JSON.stringify(feedback));
    return this.http.post<Feedback>(this.apiUrl, feedback);
  }

  getAllFeedbacksByUserId(userId: number): Observable<Feedback[]> {
    return this.http.get<Feedback[]>(this.apiUrl + '/user/' + userId);
  }

  deleteFeedback(feedbackId: number): Observable<void> {
    return this.http.delete<void>(this.apiUrl + '/' + feedbackId);
  }

  getFeedbacks(): Observable<Feedback[]> {
    return this.http.get<Feedback[]>(this.apiUrl);
  }
}

File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/springapp/src/main/java/com/examly/springapp/service/FeedbackService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.exceptions.FeedbackNotFoundException;
import com.examly.springapp.model.Feedback;

public interface FeedbackService {

    Feedback createFeedback(Feedback feedback);

    Feedback getFeedbackById(Long feedbackId) throws FeedbackNotFoundException;
  
    List<Feedback> getAllFeedbacks();

    boolean deleteFeedback(Long feedbackId);

    List<Feedback> getFeedbackByUserId(Long userId);

}
